sdk-core/src/main/java/care/data4life/sdk/network/ApiService.kt

```diff
@@ -0,0 +1,413 @@
+                tokenKey = try {
+                    authService.refreshAccessToken(alias)
+                } catch (e: D4LException) {
+                    authService.clear()
+                    return response
+                }
+                request = request.newBuilder()
+                    .removeHeader(NetworkingContract.HEADER_AUTHORIZATION)
+                    .addHeader(
+                        NetworkingContract.HEADER_AUTHORIZATION,
+                        String.format(NetworkingContract.FORMAT_BEARER_TOKEN, tokenKey)
+                    ).build()
+                response.close()
+                return chain.proceed(request)
+            }
+            return response
+        }
+        return chain.proceed(request)
+    }
+
+    /**
+     * Interceptor that attaches an OAuth access token to a request.
+     *
+     *
+     * This interceptor is used for the case where the SDK client does not
+     * handle the OAuth flow itself and merely gets an access token injected.
+     * Accordingly this interceptor does not attempt to refresh the access token
+     * if the request should fail.
+     *
+     * @param chain OkHttp interceptor chain
+     * @return OkHttp response
+     * @throws IOException
+     */
+    @Throws(IOException::class)
+    private fun staticTokenIntercept(chain: Interceptor.Chain): Response {
+        var request: Request = chain.request()
+        val alias = request.header(NetworkingContract.HEADER_ALIAS)
+        val authHeader = request.headers[NetworkingContract.HEADER_AUTHORIZATION]
+        request = request.newBuilder()
+            .removeHeader(NetworkingContract.HEADER_AUTHORIZATION)
+            .removeHeader(NetworkingContract.HEADER_ALIAS)
+            .addHeader(
+                NetworkingContract.HEADER_AUTHORIZATION,
+                String.format(NetworkingContract.FORMAT_BEARER_TOKEN, staticAccessToken)
+            ).build()
+        return chain.proceed(request)
+    }
+
+    /**
+     * Full constructor.
+     *
+     *
+     * If the staticToken parameter is set to null, the SDK will handle the full OAuth flow
+     * and fetch  access and retrieval tokens, renewing the former as needed for later
+     * requests.
+     * If the a non-null staticToken is passed, the SDK will use this value as an access token.
+     * In this case, it will not dynamical fetch or renew tokens.
+     *
+     * @param authService         AuthorizationService
+     * @param environment         Deployment environment
+     * @param clientID            Client ID
+     * @param clientSecret        Client secret
+     * @param platform            Usage platform (D4L, S4H)
+     * @param connectivityService Connectivity service
+     * @param clientName          Client name
+     * @param staticAccessToken   Prefetched OAuth token - if not null, it will be used directly (no token renewal).
+     * @param debug               Debug flag
+     */
+    init {
```

Comment:
this should be at the top as it is part of the initialization of the class and should follow the general Kotlin Coding Conventions -> see [class layout](https://kotlinlang.org/docs/coding-conventions.html#source-file-organization)

Reply:
You are totally right about this...but I like not put effort into it, since this class had been already rewritten later in the current chain (I will tag the PR once it is uploaded). This step was only necessary to enable proper testing.

Reply:
Please see #139.